--- /dev/null
+++ ../lib/PhpParser/Node/Stmt/Do_.php
@@ -3,8 +3,9 @@
 namespace PhpParser\Node\Stmt;
 
 use PhpParser\Node;
+use Rector\Core\Contract\PhpParser\Node\StmtsAwareInterface;
 
-class Do_ extends Node\Stmt
+class Do_ extends Node\Stmt implements StmtsAwareInterface
 {
     /** @var Node\Stmt[] Statements */
     public $stmts;
@@ -27,7 +28,7 @@
     public function getSubNodeNames() : array {
         return ['stmts', 'cond'];
     }
-    
+
     public function getType() : string {
         return 'Stmt_Do';
     }
