--- /dev/null
+++ ../lib/PhpParser/Node/Stmt/Function_.php
@@ -4,8 +4,9 @@
 
 use PhpParser\Node;
 use PhpParser\Node\FunctionLike;
+use Rector\Core\Contract\PhpParser\Node\StmtsAwareInterface;
 
-class Function_ extends Node\Stmt implements FunctionLike
+class Function_ extends Node\Stmt implements FunctionLike, StmtsAwareInterface
 {
     /** @var bool Whether function returns by reference */
     public $byRef;
