--- /dev/null
+++ ../lib/PhpParser/Node/Expr/Closure.php
@@ -5,8 +5,9 @@
 use PhpParser\Node;
 use PhpParser\Node\Expr;
 use PhpParser\Node\FunctionLike;
+use Rector\Core\Contract\PhpParser\Node\StmtsAwareInterface;
 
-class Closure extends Expr implements FunctionLike
+class Closure extends Expr implements FunctionLike, StmtsAwareInterface
 {
     /** @var bool Whether the closure is static */
     public $static;
